relays/bin-substrate/src/chains/rialto_messages_to_millau.rs

```diff
@@ -128,13 +128,15 @@ impl SubstrateMessageLane for RialtoMessagesToMillau {
 /// Rialto node as messages source.
 type RialtoSourceClient = SubstrateMessagesSource<
 	Rialto,
+	Millau,
 	RialtoMessagesToMillau,
```

Comment:
Couldn't the lane contain all other necessary generic types? It seems it's bound with them anyway?

Reply:
This requires some refactoring - simply using associated types from lane causes `cycle detected when using associated type bounds`. But ok - I'll try to solve it here.

Reply:
No worries, let's log this and leave for a potential follow up PR.

Reply:
I've tried some attempts, but all are adding a lot of changes to this PR, so right - let's keep it for other PRs: #1033